### PR TITLE
test: apex replay debugger e2e tests

### PR DESCRIPTION
**Pull request by CristiCanizales**
### What does this PR do?
Implementation for E2E Advanced Debugging

### What issues does this PR fix or reference?
@W-12370888@, @W-12609163@

See also https://salesforce.quip.com/qVEkABuGd7ao and https://salesforce.quip.com/q3tDAvAb2Udq

### CODE REVIEW

File: test/specs/apexReplayDebugger.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    // Run SFDX: Launch Apex Replay Debugger with Last Log File
+    const workbench = await browser.getWorkbench();
+    prompt = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Launch Apex Replay Debugger with Last Log File', 1);
+
+    // Get open text editor
+    const editorView = await workbench.getEditorView();
+
+    // Get file path from open text editor
+    const activeTab = await editorView.getActiveTab();
+    if (!activeTab) {
```
Comment (jeffb-sfdc) [inline test/specs/apexReplayDebugger.e2e.ts:108]: Are lines 108-110 needed?  Won't activeTab always be "not undefined"?

File: test/specs/apexReplayDebugger.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    await utilities.createApexClassWithTest('ExampleApexClass');
+    await utilities.pause(1);
+
+    // Push source to scratch org
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Push Source to Default Scratch Org and Override Conflicts', 3);
+
+  });
+
+  step('SFDX: Turn On Apex Debug Log for Replay Debugger', async () => {
```
Comment (jeffb-sfdc) [inline test/specs/apexReplayDebugger.e2e.ts:38]: This test is failing for me on my machine.

```
Expected: true
Received: false
[chrome 102.0.5005.196 mac os x #0-0] Error: expect(received).toBe(expected) // Object.is equality
[chrome 102.0.5005.196 mac os x #0-0]
[chrome 102.0.5005.196 mac os x #0-0] Expected: true
[chrome 102.0.5005.196 mac os x #0-0] Received: false
[chrome 102.0.5005.196 mac os x #0-0]     at Context.<anonymous> (/Users/jbeeghly/src/github.com/forcedotcom/salesforcedx-vscode-automation-tests-3/test/specs/apexReplayDebugger.e2e.ts:48:41)
```

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    prompt = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Update Checkpoints in Org', 1);
+
+    // // Verify checkpoints updating results are listed on vscode's Output section
+    // const outputPanelText = await utilities.attemptToFindOutputPanelText('Apex Replay Debugger', 'Starting SFDX: Update Checkpoints in Org', 10);
+    // expect(outputPanelText).not.toBeUndefined();
+    // expect(outputPanelText).toContain('SFDX: Update Checkpoints in Org, Step 6 of 6: Confirming successful checkpoint creation');
+    // expect(outputPanelText).toContain('Ending SFDX: Update Checkpoints in Org');
+  });
+
+  step('SFDX: Turn On Apex Debug Log for Replay Debugger', async () => {
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:78]: This test is failing on my local machine.
```
[0-0] Error in "Find and Fix Bugs with Apex Replay Debugger.SFDX: Turn On Apex Debug Log for Replay Debugger"
Error: expect(received).toBe(expected) // Object.is equality

Expected: true
Received: false
    at Context.<anonymous> (/Users/jbeeghly/src/github.com/forcedotcom/salesforcedx-vscode-automation-tests-3/test/specs/trailApexReplayDebugger.e2e.ts:88:41)
```

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+  InputBox,
+  QuickOpenBox,
+  TextEditor
+} from 'wdio-vscode-service';
+import {
+  ScratchOrg
+} from '../scratchOrg';
+import * as utilities from '../utilities';
+
+describe('Find and Fix Bugs with Apex Replay Debugger', async () => {
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:18]: Since this test suite mimics that Trailhead module, I think '"Find and Fix Bugs with Apex Replay Debugger" Trailhead module' might be better.

In addition to changing the description, maybe add a comment about how this test suite walks through the same steps performed in that module, and maybe add a link to the module as well.

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    await textEditor.moveCursor(8, 5);
+
+    // Run SFDX: Toggle Checkpoint.
+    prompt = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Toggle Checkpoint', 1);
+
+    // Run SFDX: Update Checkpoints in Org.
+    prompt = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Update Checkpoints in Org', 1);
+
+    // // Verify checkpoints updating results are listed on vscode's Output section
+    // const outputPanelText = await utilities.attemptToFindOutputPanelText('Apex Replay Debugger', 'Starting SFDX: Update Checkpoints in Org', 10);
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:72]: Is this code needed or can it be removed?

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+    // Run SFDX: Turn On Apex Debug Log for Replay Debugger
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Turn On Apex Debug Log for Replay Debugger', 1);
+
+    // Wait for the command to execute
+    await utilities.waitForNotificationToGoAway(workbench, 'Running SFDX: Turn On Apex Debug Log for Replay Debugger', fiveMinutes);
+
+    // Look for the success notification that appears which says, "SFDX: Turn On Apex Debug Log for Replay Debugger successfully ran".
+    const successNotificationWasFound = await utilities.notificationIsPresent(workbench, 'SFDX: Turn On Apex Debug Log for Replay Debugger successfully ran');
+    expect(successNotificationWasFound).toBe(true);
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:88]: This test fails on my local machine.

```
[chrome 102.0.5005.196 mac os x #0-0] expect(received).toBe(expected) // Object.is equality

Expected: true
Received: false
[chrome 102.0.5005.196 mac os x #0-0] Error: expect(received).toBe(expected) // Object.is equality
[chrome 102.0.5005.196 mac os x #0-0]
[chrome 102.0.5005.196 mac os x #0-0] Expected: true
[chrome 102.0.5005.196 mac os x #0-0] Received: false
[chrome 102.0.5005.196 mac os x #0-0]     at Context.<anonymous> (/Users/jbeeghly/src/github.com/forcedotcom/salesforcedx-vscode-automation-tests-3/test/specs/trailApexReplayDebugger.e2e.ts:88:41)
```

File: test/specs/trailApexReplayDebugger.e2e.ts (hunk 5 above)
Comment (CristiCanizales) [reply test/specs/trailApexReplayDebugger.e2e.ts:72]: Oh right, this. It's commented because I'm not being able to open the Apex Replay Debugger Output tab. It's like it can't be found. We were going to chat about it a week ago but then I completely forgot. The trailhead doesn't mention to check the output tab to verify the checkpoints were indeed set, but I thought it would be good to do so

File: test/specs/apexReplayDebugger.e2e.ts (hunk 1 above)
Comment (CristiCanizales) [reply test/specs/apexReplayDebugger.e2e.ts:108]: I think I had that because I was getting like this tslint error that activeTab might be undefined, but yeah, it's not needed. Thanks!

File: test/specs/apexReplayDebugger.e2e.ts (hunk 2 above)
Comment (jeffb-sfdc) [reply test/specs/apexReplayDebugger.e2e.ts:37]: When I run, I get the following when `SFDX: Turn On Apex Debug Log for Replay Debugger` runs:
image

sfdx version is:
sfdx-cli/7.190.2 darwin-x64 node-v18.14.1

Comment (jeffb-sfdc) [reply test/specs/apexReplayDebugger.e2e.ts:37]: I thought it might have something to do with my version ofn sfdx.  I just updated it and it's now 7.193.2 but it is still failing to run.  For what it's worth, I opened the same project out side of running with wdio, and `Turn On Apex Debug Log for Replay Debugger` worked.

Comment (CristiCanizales) [reply test/specs/apexReplayDebugger.e2e.ts:37]: It's weird. It's passing just fine on my machine...

Comment (CristiCanizales) [reply test/specs/apexReplayDebugger.e2e.ts:37]: was... #flapper

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+    // Run SFDX: Launch Apex Replay Debugger with Current File
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Launch Apex Replay Debugger with Current File', 1);
+    const successNotificationWasFound = await utilities.notificationIsPresent(workbench, 'SFDX: Launch Apex Replay Debugger with Current File successfully ran');
+    if (successNotificationWasFound !== true) {
+      const failureNotificationWasFound = await utilities.notificationIsPresent(workbench, 'You can only run this command with Anonymous Apex files, Apex Test files, or Apex Debug Log files.');
+      if (failureNotificationWasFound === true) {
+        expect(successNotificationWasFound).toBe(false);
+      } else {
+        utilities.log('Warning - Launching Apex Replay Debugger with Current File failed, neither the success notification or the failure notification was found.');
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:157]: When I run, the "SFDX: Launch Apex Replay Debugger with Current File" command does not display any notifications, and the "Warning - Launching Apex Replay Debugger with Current File failed, neither the success notification or the failure notification was found" condition was hit.

@CristiCanizales when you run, do you get this? (hit line 157 and get the warning)

If you don't get any notification (like me) I suggest changing the code to:

```
  step('Replay an Apex Debug Log', async () => {
    // Run SFDX: Launch Apex Replay Debugger with Current File
    const workbench = await browser.getWorkbench();
    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Launch Apex Replay Debugger with Current File', 1);

    // Continue with the debug session
    await browser.keys(['F5']);
    await utilities.pause(1);
    await browser.keys(['F5']);
    await utilities.pause(1);

    // TODO:
    // add expectations here
  });
```

Comment (CristiCanizales) [reply test/specs/trailApexReplayDebugger.e2e.ts:157]: @jeffb-sfdc I do get a notification, the failure one. That's why I kept it in this case so when the command fully works I just get rid of the failure check part

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+    // Wait for the command to execute
+    await utilities.waitForNotificationToGoAway(workbench, 'Running SFDX: Run Apex Tests', fiveMinutes);
+    await utilities.waitForNotificationToGoAway(workbench, 'Running SFDX: Run Apex Tests: Listening for streaming state changes...', fiveMinutes);
+    await utilities.waitForNotificationToGoAway(workbench, 'Running SFDX: Run Apex Tests: Processing test run', fiveMinutes, false);
+
+    const successNotificationWasFound = await utilities.notificationIsPresent(workbench, 'SFDX: Run Apex Tests successfully ran');
+    expect(successNotificationWasFound).toBe(true);
+
+    // Verify test results are listed on vscode's Output section
+    const outputPanelText = await utilities.attemptToFindOutputPanelText('Apex', '=== Test Results', 10);
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:118]: In tests that call `attemptToFindOutputPanelText()` (like here at line 118), could you add code which clears the output first, before the test runs?  So on 104, ad something like:
```
const outputView = await utilities.openOutputView();
await outputView.clear();
```

Also, this should be done not only here, but any other test that validates the output.

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+    testSetup = new TestSetup('TrailApexReplayDebugger', false);
+    await testSetup.setUp();
+
+    // Create Apex class AccountService
+    await utilities.createApexClassWithBugs();
+
+    // Push source to org
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Push Source to Default Org and Override Conflicts', 5);
+  });
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:38]: How about adding a check for a notification of "SFDX: Push Source to Default Org and Override Conflicts successfully ran"

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+    const outputPanelText = await utilities.attemptToFindOutputPanelText('Apex', '=== Test Results', 10);
+    expect(outputPanelText).not.toBeUndefined();
+    expect(outputPanelText).toContain('Assertion Failed: incorrect ticker symbol');
+    expect(outputPanelText).toContain('Expected: CRM, Actual: SFDC');
+  });
+
+  step('SFDX: Get Apex Debug Logs', async () => {
+    // Run SFDX: Get Apex Debug Logs
+    const workbench = await browser.getWorkbench();
+    prompt = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Get Apex Debug Logs', 1);
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:127]: output text needs to be cleared b/f the step runs

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+    await utilities.pause(1);
+
+    // Push source to org
+    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Push Source to Default Org and Override Conflicts', 6);
+  });
+
+  step('Run Apex Tests to Verify Fix', async () => {
+    // Run SFDX: Run Apex tests.
+    const workbench = await browser.getWorkbench();
+    prompt = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Run Apex Tests', 1);
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:183]: output text needs to be cleared b/f the step runs

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+  step('Set Breakpoints and Checkpoints', async () => {
+    // Get open text editor
+    const workbench = await browser.getWorkbench();
+    const editorView = workbench.getEditorView();
+    const textEditor = await editorView.openEditor('AccountService.cls') as TextEditor;
+    await textEditor.moveCursor(8, 5);
+
+    // Run SFDX: Toggle Checkpoint.
+    prompt = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Toggle Checkpoint', 1);
+
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:73]: How about adding a check here to see if the checkpoint is present?
The CSS class is `'.codicon-debug-breakpoint-conditional'`

In a browser one would do:
```
const breakpoints = document.querySelectorAll('.codicon-debug-breakpoint-conditional');
```
...and then verify that `breakpoints.length === 1`

The syntax for wdio/selenium is a little different. (the global `document` object isn't available)

I'm not 100% positive, but I think something like:
```
const breakpoints = $$('.codicon-debug-breakpoint-conditional');
expect(breakpoints.length).toEqual(1);
```

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+    // Calling SFDX: Update Checkpoints in Org fails if the org has been recently created, 
+    // it does not complete the 6 steps but only 4.
+    // Reloading the window forces the extensions to be reloaded and this seems to fix the issue.
+    await utilities.runCommandFromCommandPrompt(workbench, 'Developer: Reload Window', 10);
+    await utilities.pause(10);
+
+    // Run SFDX: Update Checkpoints in Org.
+    prompt = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Update Checkpoints in Org', 2);
+    // // Verify checkpoints updating results are listed on vscode's Output section
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:82]: There is a double comment at line 82

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+    // const outputPanelText = await utilities.attemptToFindOutputPanelText('Apex Replay Debugger', 'Starting SFDX: Update Checkpoints in Org', 10);
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:83]: lines 83-86 can be added back in

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+    prompt = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Toggle Checkpoint', 1);
+
+    // Calling SFDX: Update Checkpoints in Org fails if the org has been recently created, 
+    // it does not complete the 6 steps but only 4.
+    // Reloading the window forces the extensions to be reloaded and this seems to fix the issue.
+    await utilities.runCommandFromCommandPrompt(workbench, 'Developer: Reload Window', 10);
+    await utilities.pause(10);
+
+    // Run SFDX: Update Checkpoints in Org.
+    prompt = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Update Checkpoints in Org', 2);
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:81]: The pause here at line 81 needs to be a lot longer.  Maybe 10 seconds?  on my machine, updating the checkpoints takes quite a while.

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 16]
```diff
@@ -0,0 +1,209 @@
...
+    expect(quickPicks).not.toBeUndefined();
+    expect(quickPicks.length).toBeGreaterThanOrEqual(1);
+    await prompt.selectQuickPick('User User - ApexTestHandler');
+
+    // Wait for the command to execute
+    await utilities.waitForNotificationToGoAway(workbench, 'Running SFDX: Get Apex Debug Logs', fiveMinutes);
+
+    const successNotificationWasFound = await utilities.notificationIsPresent(workbench, 'SFDX: Get Apex Debug Logs successfully ran');
+    expect(successNotificationWasFound).toBe(true);
+  });
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:149]: At this point there is a log file that's opened, and there is text in the output tab.

While we don't control what's in the file or what's in the output tab, we should try to check for a few things.

I suggesting looking at the contents of the log file, and look for:
`|EXECUTION_STARTED`
`|EXECUTION_FINISHED`

Likewise, in the output panel. I suggest looking for:
`|EXECUTION_STARTED`
`|EXECUTION_FINISHED`
`ended SFDX: Get Apex Debug Logs`

File: test/specs/apexReplayDebugger.e2e.ts [hunk 17]
```diff
@@ -0,0 +1,179 @@
...
+    await browser.keys(['F5']);
+    await utilities.pause(1);
+  });
+
+  step('SFDX: Launch Apex Replay Debugger with Current File', async () => {
+    // Run SFDX: Launch Apex Replay Debugger with Current File
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Launch Apex Replay Debugger with Current File', 1);
+
+    const successNotificationWasFound = await utilities.notificationIsPresent(workbench, 'SFDX: Launch Apex Replay Debugger with Current File successfully ran');
```
Comment (jeffb-sfdc) [inline test/specs/apexReplayDebugger.e2e.ts:134]: I verified this on my machine while testing and against production.
The `SFDX: Launch Apex Replay Debugger with Current File` command does not display any notifications...
Are you seeing notifications on your machine?

Lines 130-145 should be removed. and in its place:
```
    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Launch Apex Replay Debugger with Current File', 1);
    await browser.keys(['F5']);
    await utilities.pause(1);

    await browser.keys(['F5']);
    await utilities.pause(1);

    // also do this:
    // switch to Debug Console
    // in the Debug look for "launchRequest: args={"name":"Launch Apex Replay Debugger","type":"apex-replay","request":"launch","
    // and look for |EXECUTION_STARTED
    // and look for |EXECUTION_FINISHED
    // and look for Apex Replay Debugger session terminated.
```

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 18]
```diff
@@ -0,0 +1,228 @@
...
+    await textEditor.moveCursor(8, 5);
+
+    // Run SFDX: Toggle Checkpoint.
+    prompt = await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Toggle Checkpoint', 1);
+
+    // Calling SFDX: Update Checkpoints in Org fails if the org has been recently created, 
+    // it does not complete the 6 steps but only 4.
+    // Reloading the window forces the extensions to be reloaded and this seems to fix the issue.
+    await utilities.runCommandFromCommandPrompt(workbench, 'Developer: Reload Window', 10);
+    await utilities.pause(15);
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:82]: OK, after chatting with you earlier, I spent 3 hours looking it this.  Here is what I found:
* I changed line 82 to 15 seconds, and line 89 to 5 seconds.
This ran successfully once
The second time it ran it failed (same error we were seeing before)

* I then changed line 82 back to 10, and line 89 to 10
This ran successfully once
The second time it ran it failed, but this time on line 191

* I change 191 from 1 to 2
This ran successfully three times
Then  on the fourth time, it failed at line 85 (not 89, but 85)

* I then changed line 82 from 10 to 15
This ran successfully once
The second time it ran it failed on line 85 again

* I then changed line 82 from 10 to 20
This ran successfully once
The second time it ran it failed on line 85 again
...and this ran successfully 6 times and I called it good.

That might have been hard to follow, so here's a summary of changes needed to work on my machine:
line 82: pause of 20
line 89: pause of 10
line 191: pause of 2

File: test/specs/trailApexReplayDebugger.e2e.ts [hunk 19]
```diff
@@ -0,0 +1,228 @@
...
+    // Run SFDX: Launch Apex Replay Debugger with Current File
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Launch Apex Replay Debugger with Current File', 5);
+
+    // Continue with the debug session
+    await browser.keys(['F5']);
+    await utilities.pause(1);
+    await browser.keys(['F5']);
+    await utilities.pause(1);
+  });
```
Comment (jeffb-sfdc) [inline test/specs/trailApexReplayDebugger.e2e.ts:182]: After the debugging finishes (line 181), I think the following should be checked:

switch to Debug Console
in the Debug console, look for `"launchRequest: args={"name":"Launch Apex Replay Debugger","type":"apex-replay","request":"launch","`
also look for `|EXECUTION_STARTED`
also look for `|EXECUTION_FINISHED`
also look for `Apex Replay Debugger session terminated.`

Comment (jeffb-sfdc) [reply test/specs/trailApexReplayDebugger.e2e.ts:182]: For getting the text of the debug console tab, as you've seen `debugConsoleView.getText()` doesn't work.  You will need to implement something like `getOutputViewText()` (in test/utilities/outputView.ts) but for the debug console.

I suggest creating a new file, /utilities/debugConsolView.ts, and add a `getDebugConsoleViewText()` function.

Comment (jeffb-sfdc) [reply test/specs/trailApexReplayDebugger.e2e.ts:182]: Darn.  Well, after discussing with Cristi, it turns out that:
* one can't call debugConsoleView.getText() (due to a bug in wdio)
* and the debug console doesn't support select all like the output panel does, so we can't implement a solution like the one that's in OutputView

File: test/specs/apexReplayDebugger.e2e.ts [hunk 20]
```diff
@@ -0,0 +1,173 @@
...
+    await utilities.createApexClassWithTest('ExampleApexClass');
+
+    // Push source to org
+    const workbench = await browser.getWorkbench();
+    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Push Source to Default Org and Override Conflicts', 1);
+    // Wait for the command to execute
+    await utilities.waitForNotificationToGoAway(workbench, 'Running SFDX: Push Source to Default Org and Override Conflicts', fiveMinutes);
+    const successPushNotificationWasFound = await utilities.notificationIsPresent(workbench, 'SFDX: Push Source to Default Org and Override Conflicts successfully ran');
+    expect(successPushNotificationWasFound).toBe(true);
+
```
Comment (jeffb-sfdc) [inline test/specs/apexReplayDebugger.e2e.ts:38]: nit pick: blank line on 38 should be removed

File: test/specs/apexReplayDebugger.e2e.ts [hunk 21]
```diff
@@ -0,0 +1,173 @@
...
+    const quickPicks = await prompt.getQuickPicks();
+    expect(quickPicks).not.toBeUndefined();
+    expect(quickPicks.length).toBeGreaterThanOrEqual(1);
+    await prompt.selectQuickPick('User User - Api');
+
+    // Wait for the command to execute
+    await utilities.waitForNotificationToGoAway(workbench, 'Running SFDX: Get Apex Debug Logs', fiveMinutes);
+
+    const successNotificationWasFound = await utilities.notificationIsPresent(workbench, 'SFDX: Get Apex Debug Logs successfully ran');
+    expect(successNotificationWasFound).toBe(true);
```
Comment (jeffb-sfdc) [inline test/specs/apexReplayDebugger.e2e.ts:102]: the output panel should be cleared (line 87)
and then add validation of some of the contents in the .log file here,
and validation of some of the contents in the output panel

File: test/specs/apexReplayDebugger.e2e.ts [hunk 22]
```diff
@@ -0,0 +1,173 @@
...
+    // Create anonymous apex file
+    await utilities.createAnonymousApexFile();
+
+    // Run SFDX: Launch Apex Replay Debugger with Editor Contents", using the Command Palette.
+    await utilities.runCommandFromCommandPrompt(workbench, 'SFDX: Execute Anonymous Apex with Editor Contents', 1);
+
+    // Wait for the command to execute
+    await utilities.waitForNotificationToGoAway(workbench, 'Running Execute Anonymous Apex', fiveMinutes);
+    const successNotificationWasFound = await utilities.notificationIsPresent(workbench, 'Execute Anonymous Apex successfully ran');
+    expect(successNotificationWasFound).toBe(true);
```
Comment (jeffb-sfdc) [inline test/specs/apexReplayDebugger.e2e.ts:154]: the output panel should be cleared (line 144)
and then add validation of some of the contents in the output panel

**Review by jeffb-sfdc — APPROVED** [review]
